### PR TITLE
migrate vue config

### PR DESCRIPTION
**Pull request by strawberry-choco**

### CODE REVIEW

File: packages/eslint-plugin-vue/src/index.ts [hunk 1]
```diff
@@ -1,2 +1,28 @@
...
+        extends: [
+            ...cloudflightTypescriptConfig,
+            // .d.ts file for eslint-plugin-vue has been implemented but not released yet, if the
+            // following line fails to build after updating the version, simply remove this comment
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access,@typescript-eslint/consistent-type-assertions
+            ...pluginVue.configs['flat/recommended'] as TSESLint.FlatConfig.ConfigArray,
+            ...vueTsEslintConfig(),
+        ],
+        rules: {
+            ...typescriptRules,
```
[the remark this answers is not in the document]
Comment (strawberry-choco) [reply packages/eslint-plugin-vue/src/index.ts:24]: Vue configs get applied to `.vue` files only, yes. Typescript config gets applied to both `.ts` (and `.js` and their variants) files and `.vue` files. Line 12 applies our typescript config as is.

File: package.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
         "@typescript-eslint/eslint-plugin": "8.8.0",
         "@typescript-eslint/parser": "8.8.0",
         "@typescript-eslint/utils": "8.8.0",
-        "@vue/eslint-config-typescript": "13.0.0",
+        "@vue/eslint-config-typescript": "14.0.0-rc.1",
```
[the remark this answers is not in the document]
Comment (strawberry-choco) [reply package.json]: Fixed.

File: packages/eslint-plugin-vue/package.json [hunk 3]
```diff
@@ -22,20 +22,18 @@
...
-    "private": true,
     "scripts": {
-        "build-packagex": "tsc"
+        "build-package": "tsc"
     },
     "dependencies": {
-        "@vue/eslint-config-typescript": "^13.0.0",
-        "eslint-plugin-vue": "^9.25.0"
+        "@cloudflight/eslint-plugin-typescript": "workspace:*",
+        "@vue/eslint-config-typescript": "^14.0.0-rc.1",
```
[the remark this answers is not in the document]
Comment (strawberry-choco) [reply packages/eslint-plugin-vue/package.json]: Fixed.

File: packages/eslint-plugin-vue/src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakob0fischl) [reply packages/eslint-plugin-vue/src/index.ts:24]: Yes, but don't you also want to disable no-unsafe-assignment in .ts files? Can you even import Vue components there? I have no idea, but seems to me it was like that before

[the remark this answers is not in the document]
Comment (strawberry-choco) [reply packages/eslint-plugin-vue/src/index.ts:24]: Vue.js components should not be imported there.

[the remark this answers is not in the document]
Comment (strawberry-choco) [reply packages/eslint-plugin-vue/src/index.ts:24]: They are imported in other `.vue` files so they can be used in the template.

**Pull request merged**